### ISSUE TITLE
How to update to scala 2.13 or 3?

### ISSUE THREAD

**Issue by mrt181** [exercism/scala#684]
What is needed to update to Scala 2.13 or 3 and make it work with exercism.
I am sure someone will do that in a PR but its likely a little more involved than just updating the build.sbt in each exercise?
The sbt-test-runner must be updated too?
Can you provide some directions?

**Comment by ErikSchierboom** [comment]
There are a couple of things that need to be done:

- The [exercises](https://github.com/exercism/scala/tree/main/exercises) should be updated to work with Scala 3
- The [test generators](https://github.com/exercism/scala/tree/main/testgen) should be updated to work with Scala 3 and output Scala 3 code
- The [docs](https://github.com/exercism/scala/tree/main/docs) should be updated
- The [scala test-runner](https://github.com/exercism/scala-test-runner) needs to support Scala 3

**Comment by bobahop** [comment]
I'm guessing this a work in progress?

I downloaded `isogram` via CLI and the `build.sbt` was set to `scalaVersion := "2.12.8"`. The Metals extension in vscode reported that 2.12.8 may not work in future versions of Metals and that Scala should be updated to at least 2.12.12. I jumped through some hoops, some of them probably unnecessary, and after being prompted to download/import a couple files the `build.sbt` was set to `scalaVersion := "2.12.14"`.

I then looked at several previously downloaded Scala exercises and they all had `scalaVersion := "2.12.14"` already.

Just to inform you that updating the Scala version is appreciated and, at least for the vscode Metals extension, somewhat needed.

**Comment by ErikSchierboom** [comment]
@luigip That sounds great! Would you be willing to submit a PR?

**Comment by LarsWestergren** [comment]
I'm willing to try this.

* Exercises 

Warnings and errors with sbt 1.71. and Scala 3.1.3

- [ ] alphametics
- [ ] armstrong-numbers
- [ ] binary-search
- [ ] binary-search-tree
- [ ] complext-numbers
- [ ] forth
- [ ] grade-school
- [ ] lens-person
- [ ] linked-list
- [ ] matching-brackets
- [ ] matrix
- [ ] nth-prime
- [ ] octal
- [ ] pangram
- [ ] say
- [ ] sgf-parsing
- [ ] sieve
- [ ] simple-linked-list
- [ ] spiral-matrix
- [ ] trinary
- [ ] word-count
- [ ] wordy

* Test Generators

* Docs

* scala test-runner project

**Comment by ErikSchierboom** [comment]
@LarsWestergren Thanks a ton!

**Comment by asarkar** [comment]
Hi, I'm planning to start the Scala track and wanted to check if there is any interest in moving forward with this ticket.

**Comment by ErikSchierboom** [comment]
Definitely! I see that you also posted at https://github.com/exercism/scala/pull/734#issuecomment-1868043660, which is great!
